src/hooks/useTrackListSearchParams.ts

```diff
@@ -51,51 +52,62 @@ function isValidSearchParameter<T>(value: unknown, options: T[]): value is T {
   return options.includes(value as T);
 }
 
-function getValidatedSearchParam<T>(
+function getNumberParam(
   searchParams: URLSearchParams,
   key: string,
-  validator: (value: unknown) => value is T,
-): T | null;
-
-function getValidatedSearchParam<T>(
-  searchParams: URLSearchParams,
-  key: string,
-  validator: (value: unknown) => value is T,
-  defaultValue: T,
-): T;
+  defaultValue: number,
+) {
+  return pipe(
+    O.fromNullable(searchParams.get(key)),
+    O.flatMap((value) => {
+      const n = Number(value);
+      return isNaN(n) ? O.None : O.Some(n);
```

Comment:
[isNaN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/isNaN) converts the argument to a Number and returns true if the resulting value is NaN.
[Number.isNaN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN) does not convert the argument; it returns true when the argument is a Number and is NaN.

```suggestion
      return Number.isNaN(n) ? O.None : O.Some(n);
```

Reply:
Makes sense, fixed